### ISSUE TITLE
TDD the Knight

### ISSUE THREAD

**Issue by programarivm** [chesslablab/ts-chess#13]
At this stage everything's ready to start writing the chess pieces!

Remember:

> We're writing an object-oriented chess codebase not only because we want to practice software development skills but also to avoid the intricacies of chess bitboards.

The transcription from [PHP Chess](https://github.com/chesslablab/php-chess) to [TypeScript Chess](https://github.com/chesslablab/ts-chess) is going well, however, it is worth saying that is quite a challenge that needs to be completed accurately. 

Let's stick to the following guidelines for the time being:

- Add one chess piece at a time
- Implement the `mobility()` method only
- Forget about `AbstractPiece`

In other words, for the sake of simplicity the next step consists in adding all the chess pieces to the codebase with the condition that only the `mobility()` method is implemented for the time being. Then, we'll be in a better position to start coding `AbstractPiece`.

Thus, this issue is about TDDing the `mobility()` method in a new .ts file called `Knight.ts` in the [ts-chess/src/piece](https://github.com/chesslablab/ts-chess/tree/master/src/piece) folder.

For further information, please visit:

- [What Does It Mean to TDD a Class?](https://bootcamp.uxdesign.cc/what-does-it-mean-to-tdd-a-class-4dbb9bd6b5e2)

Keep it up and happy coding and learning!

**Comment by owais34** [comment]
Hi Please assign me this issue
